docs/src/modules/multilayerqg.md

```diff
@@ -1,5 +1,85 @@
 # MultiLayerQG
 
+
+### From continuous to discrete
+
+Under the quasi-geostrophic approximation with stratified dynamics both the horizontal flow, ``u(x, y, z, t)``, 
+``v(x, y, z, t)``, and the buoyancy ``b(x, y, z, t)`` are obtained via a streamfunction ``\psi(x, y, z, t)``:
+
+```math
+\begin{aligned}
+  u &=   - \partial_y \psi , \\
+  v &=     \partial_x \psi , \\
+  b &= f_0 \partial_z \psi .
+\end{aligned}
+```
+
+The dynamics are determined from the quasi-geostrophic potential vorticity (QGPV)
+
+```math
+\begin{aligned}
+  q &= \partial_x v - \partial_y u + \partial_z \left( \frac{f_0}{N^2} b \right) , \\
+    &= (\partial_x^2 + \partial_y^2) \psi + \partial_z \left( \frac{f_0^2}{N^2} \partial_z \psi \right) .
+\end{aligned}
+```
+
+where ``N^2 = - \rho_0^{-1} g \, \mathrm{d} \rho / \mathrm{d}z`` is the square of the Brunt-Väisälä
+frequency for a Boussinesq fluid, where ``\rho_0`` is a reference density and ``g`` the gravitational
+constant.
+
+To go from the continuous (in ``z``) dynamics above to a discrete dynamics we can integrate all
+quantities over the height of each fluid layer.
+An alternative is to replace the vertical derivatives that appear both in the buoyancy definition and
+in the Brunt-Väisälä frequency with a discrete approximation.
+
+We use subscripts ``j`` to denote quantities that are constant within each fluid layer. For example,
+the heights of each fluid layer are ``H_j``, ``j = 1, \dots, n``, where ``n`` is the number of fluid
+layers. Also, ``\rho_j`` is the density of each layer and ``\psi_j`` and ``q_j`` the streamfunction 
+and QGPV associated with that fluid layer.
+
+The buoyancy is associated with the fluid interfaces. We use semi-integer subscript values to denote a
+fluid interface, e.g., ``b_{5/2}`` is the buoyancy that corresponds to the interface between fluid layers
+2 and 3. Thus, from the relationship that connects buoyancy ``b`` and streamfunction ``\psi`` we approximate
+the buoyancies at each interface as
```

Comment:
This explanation suggests that we are really doing a transformation into buoyancy coordinates.

I'd suggest doing the transformation in continuous space first, ie write down QG in buoyancy coordinates. _Then_ you can introduce a "finite volume" approximation to QG in buoyancy coordinates. Since `b` is the coordinate, a finite volume discretization of `b` would indeed identify a buoyancy value for each layer.